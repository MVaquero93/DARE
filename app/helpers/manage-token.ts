import axios from 'axios'
import envConfig from '../config/env'
import { TokenData } from '../interfaces/tokenData.interface'
import { login } from '../services/login.service'
let data = {} as TokenData

export const getToken = async () => {
  return data
}

export const setToken = async () => {
  try {
    data = await login()

    setTimeout(() => {
      data = {} 
    }, envConfig.credentials.expires_in)

    return data
  } catch (err) {
    console.error(err)
    return null
  }
}