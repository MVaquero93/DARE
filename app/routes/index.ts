import express from 'express'
import apiRouter from './api'


const router = express.Router()

/**
 * load api routes
 */
router.use('/api', apiRouter)


export default router
