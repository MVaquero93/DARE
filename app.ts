import express from 'express'
import envConfig from './app/config/env'
import routes from './app/routes'
import bodyParser from 'body-parser'

const app = express()


app.use(
  bodyParser.urlencoded({extended: false}),
  bodyParser.json(),
)
app.use(routes)

app.get('/', (req, res) => {
  res.send('Welcome to DARE')
})

const server = app.listen(envConfig.app.port, () => {
  console.log('listening at port: ' + envConfig.app.port)
})

export default server
