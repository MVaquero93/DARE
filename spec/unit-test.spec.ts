import { setToken } from '../app/services/manage-token.service'
import app from '../app'

describe('unit testing example', () => {

  afterAll((done) => {
    app.close(() => {
      done()
    })
  });

  it('Set token method', async () => {
    const data = await setToken()
    data.token = ''
    expect(data).toMatchSnapshot()
  })
})