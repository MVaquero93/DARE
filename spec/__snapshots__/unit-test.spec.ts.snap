// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit testing example Set token method 1`] = `
Object {
  "expires_in": 200,
  "token": "",
  "type": "Bearer",
}
`;
